src/core/zowe/core_for_zowe_sdk/config_file.py

```diff
@@ -348,3 +347,145 @@ def load_profile_properties(self, profile_name: str) -> dict:
+        current_profile["secure"] = current_secure
+        self.__set_or_create_nested_profile(profile_name, current_profile)
+
+    def set_profile(self, profile_path: str, profile_data: dict) -> None: 
+        """
+        Set a profile in the config file.
+
+        Parameters:
+            profile_path (str): The path of the profile to be set. eg: profiles.zosmf
+            profile_data (dict): The data to be set for the profile.
+        """
+        if self.profiles is None:
+            self.init_from_file()
+        profile_name = self.get_profile_name_from_path(profile_path)
+        if "secure" in profile_data:
+            # Checking if the profile has a 'secure' field with values
+            secure_fields = profile_data["secure"]
+            current_profile  = self.find_profile(profile_name,self.profiles) or {}
+            existing_secure_fields = current_profile.get("secure", [])
+            new_secure_fields = [field for field in secure_fields if field not in existing_secure_fields]
+
+            # JSON paths for new secure properties and store their values in CredentialManager.secure_props
+            for field in new_secure_fields:
+                json_path = f"{profile_path}.properties.{field}.value"
+                CredentialManager.secure_props[self.filepath] = {
+                    **CredentialManager.secure_props.get(self.filepath, {}),
+                    json_path: profile_data["properties"][field]
+                }
+
+            # Updating the 'secure' field of the profile with the combined list of secure fields
+            profile_data["secure"] = existing_secure_fields + new_secure_fields
+            # If a field is provided in the 'secure' list and its value exists in 'profile_data', remove it
+            profile_data["properties"] = {
+                field: value
+                for field, value in profile_data.get("properties", {}).items()
+                if field not in profile_data["secure"]
+            }
+        self.__set_or_create_nested_profile(profile_name, profile_data)
+        
+
+    def save(self, secure_props=True):
+        """
+        Save the config file to disk. and secure props to vault
+        parameters:
+            secure_props (bool): If True, the secure properties will be stored in the vault. Default is False.
+        Returns:
+            None
+        """
+        # Update the config file with any changes
+        if self.profiles is None:
+            self.init_from_file()
+        elif any(self.profiles.values()):
+            with open(self.filepath, 'w') as file:
+                # Update the profiles in the JSON data
+                self.jsonc["profiles"] = self.profiles
+                file.seek(0)  # Move the file pointer to the beginning of the file
+                commentjson.dump(self.jsonc, file, indent=4)
+                file.truncate()  # Truncate the file to the current file pointer position
```

Comment:
Are the `seek` and `truncate` calls necessary?

I believe when a file is opened in "w" mode any contents are deleted so the pointer is already at 0.